CMakeLists.txt

```diff
@@ -651,35 +652,57 @@ if(Iconv_FOUND)
 	target_link_libraries(torrent-rasterbar PRIVATE Iconv::Iconv)
 endif()
 
-find_public_dependency(OpenSSL)
-set_package_properties(OpenSSL
-	PROPERTIES
-		URL "https://www.openssl.org/"
-		DESCRIPTION "Full-strength general purpose cryptography library"
-		TYPE RECOMMENDED
-		PURPOSE "Provides HTTPS support to libtorrent"
-)
-
-if(OPENSSL_FOUND)
+if(NOT gnutls)
+	find_public_dependency(OpenSSL)
+	set_package_properties(OpenSSL
+		PROPERTIES
+			URL "https://www.openssl.org/"
+			DESCRIPTION "Full-strength general purpose cryptography library"
+			TYPE RECOMMENDED
+			PURPOSE "Provides HTTPS support to libtorrent"
+	)
+	if(OPENSSL_FOUND)
+		# TODO: needed until https://gitlab.kitware.com/cmake/cmake/issues/19263 is fixed
+		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
+			target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
+		endif()
 
-	# TODO: needed until https://gitlab.kitware.com/cmake/cmake/issues/19263 is fixed
-	if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
-		target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
+		target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
+		target_compile_definitions(torrent-rasterbar
+			PUBLIC
+				TORRENT_USE_OPENSSL
+				TORRENT_SSL_PEERS
+				OPENSSL_NO_SSL2)
 	endif()
+endif()
 
-	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
-	target_compile_definitions(torrent-rasterbar
-		PUBLIC
-			TORRENT_USE_LIBCRYPTO
-			TORRENT_USE_OPENSSL
-			TORRENT_SSL_PEERS
-			OPENSSL_NO_SSL2)
-	target_sources(torrent-rasterbar PRIVATE src/pe_crypto)
+if(gnutls OR NOT OPENSSL_FOUND)
+	find_public_dependency(GnuTLS)
+	set_package_properties(GnuTLS
+		PROPERTIES
+			URL "https://www.gnutls.org/"
+			DESCRIPTION "GnuTLS is a free software implementation of the TLS and DTLS protocols"
+			TYPE RECOMMENDED
+			PURPOSE "Provides HTTPS support to libtorrent"
+	)
+	if(GNUTLS_FOUND)
+		target_link_libraries(torrent-rasterbar PUBLIC GnuTLS::GnuTLS)
```

Comment:
it just occurred to me that `libgcrypt` is a dependency of `libgnutls`, isn't it? Unless I've misunderstood something, it might make sense to "enable" LibGcrypt when building in gnutls mode, like how `libcrypto` is "enabled" when building in openssl mode. This is a refinement that could be done later though, as it's not obvious whether it's a good idea or what the consequences might be.

Reply:
`libgcrypt` was indeed the cryptographic backend for `libgnutls`, however they changed for `nettle` in the last versions, so there is no guarantee the libs will be present together.